packages/domain/lauréat/src/garantiesFinancières/mainlevée/consulter/consulterHistoriqueDemandeMainlevéeRejetéeGarantiesFinancières.query.ts

```diff
@@ -50,11 +52,15 @@ export const registerConsulterHistoriqueDemandeMainlevéeRejetéeGarantiesFinanc
     const identifiantProjetValueType =
       IdentifiantProjet.convertirEnValueType(identifiantProjetValue);
 
-    const result = await find<HistoriqueMainlevéeRejetéeGarantiesFinancièresEntity>(
-      `historique-mainlevee-rejetee-garanties-financieres|${identifiantProjetValueType.formatter()}`,
+    const result = await find<MainlevéeGarantiesFinancièresEntity>(
+      `mainlevee-garanties-financieres|${identifiantProjetValueType.formatter()}`,
     );
 
-    if (Option.isNone(result)) {
+    const hasNoHistoriqueMainlevée =
+      Option.isNone(result) ||
+      result.détailsMainlevées.filter((mainlevée) => mainlevée.statut === 'rejeté').length < 1;
```

Comment:
C'est toujours mieux d'utiliser le valueType

```suggestion
      result.détailsMainlevées.filter((mainlevée) => StatutMainlevéeGarantiesFinancières.convertirEnValueType(mainlevée.statut).estRejeté()).length < 1;
```

Reply:
Yes merci